FlowCrypt/Controllers/SideMenu/Main/SideMenuNavigationController.swift

```diff
@@ -96,6 +103,167 @@ final class SideMenuNavigationController: ENSideMenuNavigationController {
     }
 }
 
+// MARK: Refresh keys from EKM
+extension SideMenuNavigationController {
+    private func refreshKeysFromEKMIfNeeded(context: AppContextWithUser) {
+        Task {
+            do {
+                let configuration = try await context.clientConfigurationService.configuration
+                guard configuration.checkUsesEKM() == .usesEKM else {
+                    return
+                }
+                let emailKeyManagerApi = EmailKeyManagerApi(clientConfiguration: configuration)
+                let idToken = try await IdTokenUtils.getIdToken(userEmail: context.user.email)
+                let keydDetails = try await emailKeyManagerApi.getPrivateKeys(idToken: idToken)
+                let localKeys = try context.encryptedStorage.getKeypairs(by: context.user.email)
+                var savedPassPhrase = localKeys.first(where: { $0.passphrase != nil })?.passphrase
+                var isKeyUpdated = false
+                for keyDetail in keydDetails {
+                    guard let savedLocalKey = localKeys.first(where: { $0.primaryFingerprint == keyDetail.primaryFingerprint }) else {
+                        // No keys found in local. Add it
+                        savedPassPhrase = try await saveKeyToLocal(
+                            context: context,
+                            keyDetail: keyDetail,
+                            passPhrase: savedPassPhrase,
+                            isNewKey: true
+                        )
+                        isKeyUpdated = true
+                        continue
+                    }
```

Comment:
More obvious logic, I think, would be with two loops. 

Pseudo code

```swift
for keyDetail in keyDetals {
  if ... {
    keysToUpdate.append(keyDetail)
  }
}
if keysToUpdate.empty() {
  return
}
if passPhrase == nil {
  passPhrase = // show pass phrase modal
}
for keyToSave in keysToSave {
  saveKeyToLocal(context, keyToSave, passPhrase)
}
```

I don't think you'll need `isNewKey` in the pass phrase prompt. In either case, pass phrase should be requested for the operation as a whole (updating all account keys), not for each individual key.

Reply:
`isNewKey` is passed because after user enter passphrase, we need to check if entered passphrase matches previous key detail if it's already existing key.
 
```
// For new key just check if there are any matching keys
if isNewKey {
    return matchingKeys.isNotEmpty
}
// now figure out if the pass phrase also matched the signing prv itself
let matched = matchingKeys.first(where: { $0.fingerprints.first == keyDetail.primaryFingerprint })
return matched != nil// true if the pass phrase matched signing key
```

Reply:
That's the thing - I don't think that's needed. I'll take a closer look.

Reply:
That's the thing, I don't think we need that here.

Instead of asking for pass phrase "for a particular key", ask for a pass phrase "for the whole update opration". Then you don't have to fiddle with the `isNewKey` and the whole logic cleans up.